universal_transfer.md

```diff
@@ -0,0 +1,112 @@
+# Universal Transfer Method for Open-Attestation Documents
+
+## Status
+
+Draft
+
+## Rationale
+
+The goal of the universal transfer method is to ensure that a standard is available to all clients interacting with OA documents. This allows different clients to be itneropable with one another. The standard is extremely important if we want to provide a client like an [identity wallet](https://github.com/Open-Attestation/identity-wallet) that supports all kinds of OA document.
+
+Important considerations of the transfer method:
+
+- Reference to resource (uri)
+- Encrypted resource (key)
+- Permitted actions (termsOfUse)
+- Safe transfer (encoding)
+- Universal resolver (deep-linking)
+
+## Proposed Solution
+
+Sample QR code:
+
+![Proposed QR](assets/universal_transfer/proposed-qr.png)
+
+```
+https://openattestation.com/?resource=%7B%22uri%22%3A%22https%3A%2F%2Fsomehostedresources.com%2Fdoc%2F7de3bce4-de62-4628-914e-97d41e642582%22%2C%22enc%22%3A%7B%22type%22%3A%22OPEN-ATTESTATION-TYPE-1%22%2C%22key%22%3A%22aa57eb519fd3c63c42c2f2697e8957198b56fc945c4db18b480c07d2e6485a93%22%7D%2C%22permittedAction%22%3A%5B%22VIEW%22%2C%22STORE%22%5D%7D
+```
+
+Decoded Resource (after `?resource=`):
+
+```json
+{
+  "uri": "https://somehostedresources.com/doc/7de3bce4-de62-4628-914e-97d41e642582",
+  "enc": {
+    "type": "OPEN-ATTESTATION-TYPE-1",
+    "key": "aa57eb519fd3c63c42c2f2697e8957198b56fc945c4db18b480c07d2e6485a93"
+  },
+  "permittedAction": ["VIEW", "STORE"],
+  "redirect": "https://tradetrust.io/"
+}
+```
+
+The proposed solution is to use universal/deep links to address the namespace portion. This allow us to :
+
+1. Provide an application (think universal router) at openattestation.com to handle the action if the user scans the code using a standard QR code scanner on mobile and redirect to a specific client, represented in `redirect`.
+2. Allow any web client (tradetrust.io or opencerts.io) to scan and process the message at the path `/resource?=<encoded-json>`
+3. Provide [deep linking](https://docs.expo.io/versions/latest/workflow/linking/) opportunities for iOS/Android app to open the correct application on the phone to process the action.
+
+### Considerations
+
+1. The decryption key is now in the path instead of appearing after `#`. If the connection is not secure, the key might be leaked to mitm. 
+
+## Review on Current State
+
+Currently there is no universal standard for handling transfer of OA documents. There exist two main methods of transferring now:
+
+- QR code
+- Link
+
+### QR Code
+
+There is two implementation currently:
+
+#### TradeTrust
```

Comment:
some considerations i noted in my notebook:

why it's not https:// 
1. so that people cannot mistakenly open the url using phone camera qr code scanner
2. so that we have the option of registering a deeplink to mobile app in future
3. so that we have the option of registering a protocol handler on desktop in the future

Reply:
The alternative flow might be worse as discussed. Will add in interaction flow in this ADR to reflect the discussion.